.github/workflows/main.yml

```diff
@@ -128,9 +122,9 @@ jobs:
             use_coverage: true
 
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
```

Comment:
i beleive this one needs extra parameters to unshallow the repository as a depth of 1 results in incorrect data for setuptools_scm

Reply:
Thanks, I'll try `fetch-depth: 0`.

Actually, do you think `- run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*` as suggested [here](https://github.com/actions/checkout#fetch-all-tags) will be sufficient for setuptools_scm? I'll try that first, as presumably it would be faster than a full fetch.

Reply:
its not clear to me what works best, git shallow is very incomplete and unpolished

Reply:
Right, this doesn't work either. Will try some other things, otherwise will revert back to the 1 version.

Reply:
`git fetch --prune --unshallow` seems to have done the trick, seem to be slower by a few seconds but that's fine I think. Maybe they'll add nicer support for this at some point.

Reply:
Can confirm also using `git fetch --prune --unshallow` in other projects that use setuptools_scm (eg. https://github.com/ultrajson/ultrajson/blob/master/.github/workflows/deploy.yml#L19)